### ISSUE TITLE
Negative train and validation loss as training proceeds

### ISSUE THREAD

**Issue by cs5090237** [tensorflow/ranking#37]
For document relevance score > 1 as training proceeds, loss decreases and becomes negative. Which should not be the case. I am using **sigmoid_cross_entropy_loss**.

The NN architecture is similar to what is provided in example, with group size 2.

Could you please assist in debugging what might be going wrong?

Below is the output log during training:

INFO:tensorflow:Running training and evaluation locally (non-distributed).
INFO:tensorflow:Start train and evaluate loop. The evaluate will happen after every checkpoint. Checkpoint frequency is determined based on RunConfig arguments: save_checkpoints_steps 1000 or save_checkpoints_secs None.
INFO:tensorflow:Calling model_fn.
INFO:tensorflow:Use groupwise dnn v2.
INFO:tensorflow:Done calling model_fn.
INFO:tensorflow:Create CheckpointSaverHook.
INFO:tensorflow:Graph was finalized.
INFO:tensorflow:Running local_init_op.
INFO:tensorflow:Done running local_init_op.
INFO:tensorflow:Saving checkpoints for 0 into ./models_08Mar_T3/model.ckpt.
INFO:tensorflow:loss = 0.71311927, step = 1
INFO:tensorflow:global_step/sec: 2.5571
INFO:tensorflow:loss = 0.29717356, step = 101 (39.108 sec)
INFO:tensorflow:global_step/sec: 2.61819
INFO:tensorflow:loss = 0.18879782, step = 201 (38.194 sec)
INFO:tensorflow:global_step/sec: 2.6048
INFO:tensorflow:loss = -0.15803288, step = 301 (38.391 sec)
INFO:tensorflow:global_step/sec: 2.61858
INFO:tensorflow:loss = -0.27342975, step = 401 (38.189 sec)
INFO:tensorflow:global_step/sec: 2.62001
INFO:tensorflow:loss = -0.7961479, step = 501 (38.167 sec)
INFO:tensorflow:global_step/sec: 2.61745
INFO:tensorflow:loss = -2.0954611, step = 601 (38.205 sec)
INFO:tensorflow:global_step/sec: 2.62294
INFO:tensorflow:loss = -1.7541554, step = 701 (38.125 sec)
INFO:tensorflow:global_step/sec: 2.60777
INFO:tensorflow:loss = -2.5586243, step = 801 (38.347 sec)
INFO:tensorflow:global_step/sec: 2.62359
INFO:tensorflow:loss = -2.4577546, step = 901 (38.116 sec)

**Comment by xuanhuiwang** [comment]
This is weird. Can you make sure that you don't have negative labels in your data? Also, did you try other loss functions and observe negative loss as well?

**Comment by cs5090237** [comment]
There are no negative labels in the data.

For **sigmoid_cross_entropy_loss**, loss is negative only when relevance is > 1. For relevance = 1, loss converges nicely and is >0 all the time.

Yes, loss is also negative for **approx_ndcg_loss**.

I think there might be an issue with how the loss is calculated from the output of score_fn. Please review and help in solving this.

**Comment by xuanhuiwang** [comment]
Based on the documentation here, only 0 and 1 are valid for labels of sigmoid_cross_entropy_loss: https://cs.corp.google.com/piper///depot/google3/third_party/tensorflow/python/ops/losses/losses_impl.py?q=sigmoid_cross_entropy_loss&g=0&l=655.

How about softmax_loss? If that's ok by now, you can use that.

**Comment by xuanhuiwang** [comment]
The loss in approx_ndcg_loss is the negative of the approximate NDCG, so it should be negative by design.
